### DISCUSSION TITLE
Can I make a random edge_index by the given node list?

### DISCUSSION THREAD

**Discussion by songsong0425** [pyg-team/pytorch_geometric#9370] [Q&A] [answered]
Greetings, I have a question about the generation of random edges by the given node list for the transductive setting.
When I have a node list and want to generate the _randomly directed edges_ between them _without the self-loop_, what should I do?
(i.e., if there is pair between a-b, b-a should be removed)
The node list is as below:
```python
nodes
#tensor([    28,     38,     48,     85,    114,    144,    190,    219,    228,
#           255,    362,    386,    387,    420,    516,    529,    614,    773, ...])

result
#tensor([[387, 228, ..., 362, 3905, 6254],
#        [420, 7315, 8018, ..., 28, 529]])
```

I tried to use `torch.randint` and `torch.randperm`, but it's a little bit confusing.
Are there any pre-defined functions for this task?

**Comment by EdisonLeeeee** [comment, marked as answer]
Does that work for you?
```python
import torch
num_nodes = 100
num_edges = 100
row = torch.randint(0, num_nodes, (num_edges,))
col = torch.randint(0, num_nodes, (num_edges,))
edge_index = torch.stack([row, col], dim=0)
edge_index = edge_index[:, row < col] # make directed
```

**Reply by songsong0425** [reply]
Thank you for your answer! it works well!
